### ISSUE TITLE
GCP slurp error: TypeError in firewall.py

### ISSUE THREAD

**Issue by devlinrcg** [Netflix/security_monkey#720]
Hey everyone.
First of all, thank you for your time reading this.

I have a Security Monkey installation on AWS, with AWS accounts. I've updated this installation so I could monitor my GCP projetcs, but I'm getting the following error on securitymonkey.log:

2017-05-10 11:33:00,959 ERROR: Job "run_change_reporter (trigger: interval[0:15:00], next run at: 2017-05-10 11:47:39.009484)" raised an exception [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:520]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/apscheduler/scheduler.py", line 512, in _run_job
    retval = job.func(*job.args, **job.kwargs)
  File "/usr/local/src/security_monkey/security_monkey/scheduler.py", line 32, in run_change_reporter
    reporter.run(account, interval)
  File "/usr/local/src/security_monkey/security_monkey/reporter.py", line 56, in run
    (items, exception_map) = monitor.watcher.slurp()
  File "/usr/local/src/security_monkey/security_monkey/watchers/gcp/gce/firewall.py", line 71, in slurp
    return slurp_items()
  File "/usr/local/src/security_monkey/venv/local/lib/python2.7/site-packages/cloudaux-1.2.2-py2.7.egg/cloudaux/gcp/decorators.py", line 132, in decorated_function
    itm, exc = func(*args, **kwargs)
  File "/usr/local/src/security_monkey/security_monkey/watchers/gcp/gce/firewall.py", line 68, in slurp_items
    config=modify(rule, format='camelized')))
TypeError: modify() got an unexpected keyword argument 'format'

Don't get any information on my dashboard: zero items.

Just followed the procedures on https://github.com/Netflix/security_monkey/blob/develop/docs/update.md.
Everything else is working just fine.

Anybody else getting this?

Cheers.

**Comment by supertom** [comment]
Hi @devlinrcg, thanks for filing and sorry this happened to you. 

Looks like it was introduced with this [change](https://github.com/Netflix-Skunkworks/cloudaux/commit/56814cd42f132ebf23760dfc913e921615f7dc21#diff-b42d8e088a186f616ee760e4809cc38eR20) to cloudaux.

As to what to do next: You could downgrade cloudaux to the previous version (assuming the installed version is cloudaux 1.2.2, so you would want cloudaux 1.2.1), but there is a chance you might lose particular features you want.  The good news is that 

If you can pull from source, the change is small and quick and we can have probably have a patch for this soon.  From what I see It involves a two line change, and some version-locking to prevent it from happening to others.

**Comment by monkeysecurity** [comment]
I introduced this bug. Sorry about that.

I have a security_monkey PR that should fix this:
https://github.com/Netflix/security_monkey/pull/719/files#diff-75439b1f3287946789e3f42a1bff8213R62

This PR also does a number of other things.  I'm working to finish it up ASAP, but it's not quite ready.

**Comment by supertom** [comment]
@monkeysecurity I just sent a small PR that only addresses this issue specifically.  If your change supersedes mine, feel free to reject my PR.

**Comment by devlinrcg** [comment]
All done! Cloned the Develop branch, all working beautifully.

Thank you @supertom and @monkeysecurity.

**Comment by mikegrima** [comment]
It looks like this was broken in #723 (always results in `config = None` for GCP firewall rules):

Broken Code:
```
def modify(d, output='camelized'):
    return cloudaux_modify(d, output=format)  # 'output' should = 'output'
```

Fixing... in fix for #724 .

**Comment by mikegrima** [comment]
Fix in #742.
